numba/targets/tupleobj.py

```diff
@@ -380,3 +382,98 @@ def tuple_index_impl(tup, value):
         raise ValueError("tuple.index(x): x not in tuple")
 
     return tuple_index_impl
+
+
+#------------------------------------------------------------------------------
+#Intrinsics
+
+@intrinsic
+def make_tuple(typingctx, size, iterable):
```

Comment:
@stuartarchibald I did not put it in `numba.unsafe` because I thought it was no longer "unsafe". While it does use `tuple_setitem` which is unsafe, I though I caught the cases that would make use of `tuple_setitem` unsafe. Did I overlook something? In any case, I am happy to move it.